sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGenerator.java

```diff
@@ -95,14 +97,19 @@ private String getCondition(final Part.IgnoreCaseType ignoreCase, final String s
      * @param sqlKeyword sql key word, operation name
      * @param subject sql column name
      * @param parameter sql filter value
+     * @param takesCaseSensitiveParam if the function type can take the third boolean param
      * @return condition string
      */
     private String getFunctionCondition(final Part.IgnoreCaseType ignoreCase, final String sqlKeyword,
-                                        final String subject, final String parameter) {
+                                        final String subject, final String parameter, final boolean takesCaseSensitiveParam) {
         if (Part.IgnoreCaseType.NEVER == ignoreCase) {
             return String.format("%s(r.%s, @%s)", sqlKeyword, subject, parameter);
         } else {
-            return String.format("%s(UPPER(r.%s), UPPER(@%s))", sqlKeyword, subject, parameter);
+            if (takesCaseSensitiveParam) {
+                return String.format("%s(r.%s, @%s, true)", sqlKeyword, subject, parameter);
+            } else {
+                return String.format("%s(UPPER(r.%s), UPPER(@%s), true)", sqlKeyword, subject, parameter);
```

Comment:
```suggestion
                return String.format("%s(UPPER(r.%s), UPPER(@%s))", sqlKeyword, subject, parameter);
```

Why the true - doesn't the comment explain if teh Sql Function supports third parameter we would never get here?

Reply:
Good catch, removed!